hypershift-operator/controllers/hostedcluster/internal/platform/powervs/powervs.go

```diff
@@ -0,0 +1,291 @@
+package powervs
+
+import (
+	"context"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	k8sutilspointer "k8s.io/utils/pointer"
+	capiibmv1 "sigs.k8s.io/cluster-api-provider-ibmcloud/api/v1beta1"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/support/upsert"
+)
+
+const (
+	// TODO(mkumatag): Move to OpenShift built image
+	imageCAPIBM = "k8s.gcr.io/capi-ibmcloud/cluster-api-ibmcloud-controller:v0.2.0"
+)
+
+type PowerVS struct {
+}
+
+func (p PowerVS) DeleteCredentials(ctx context.Context, c client.Client, hcluster *hyperv1.HostedCluster, controlPlaneNamespace string) error {
+	//TODO(mkumatag): implement me
+	return nil
+}
+
+func (p PowerVS) ReconcileCAPIInfraCR(ctx context.Context, c client.Client, createOrUpdate upsert.CreateOrUpdateFN,
+	hcluster *hyperv1.HostedCluster,
+	controlPlaneNamespace string,
+	apiEndpoint hyperv1.APIEndpoint) (client.Object, error) {
+	ibmCluster := &capiibmv1.IBMPowerVSCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: controlPlaneNamespace,
+			Name:      hcluster.Name,
+		},
+	}
+
+	ibmCluster.Spec.ServiceInstanceID = hcluster.Spec.Platform.PowerVS.ServiceInstanceID
+
+	_, err := createOrUpdate(ctx, c, ibmCluster, func() error {
+		ibmCluster.Annotations = map[string]string{
+			capiv1.ManagedByAnnotation: "external",
+		}
+
+		// Set the values for upper level controller
+		ibmCluster.Status.Ready = true
+		ibmCluster.Spec.ControlPlaneEndpoint = capiv1.APIEndpoint{
+			Host: apiEndpoint.Host,
+			Port: apiEndpoint.Port,
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	// reconciliation strips TypeMeta. We repopulate the static values since they are necessary for
+	// downstream reconciliation of the CAPI Cluster resource.
+	ibmCluster.TypeMeta = metav1.TypeMeta{
+		Kind:       "IBMPowerVSCluster",
+		APIVersion: capiibmv1.GroupVersion.String(),
+	}
+	return ibmCluster, nil
+}
+
+func (p PowerVS) CAPIProviderDeploymentSpec(hcluster *hyperv1.HostedCluster, _ *hyperv1.HostedControlPlane) (*appsv1.DeploymentSpec, error) {
+	defaultMode := int32(420)
+	deploymentSpec := &appsv1.DeploymentSpec{
+		Template: corev1.PodTemplateSpec{
+			Spec: corev1.PodSpec{
+				TerminationGracePeriodSeconds: k8sutilspointer.Int64Ptr(10),
+				Tolerations: []corev1.Toleration{
+					{
```

Comment:
why this toleration?

Reply:
Seems like a copy paste thing from other providers, yes we don't need in the hypershift environment because these pods can anywhere, will remove this toleration. And we also should remove for other providers as well in a separate PR.